form/direct_submit.rst

```diff
@@ -89,3 +89,8 @@ method, pass the submitted data directly to
     "PATCH" method, the validation extension will only handle the submitted
     fields. If the underlying data needs to be validated, this should be done
     manually, i.e. using the validator.
+    
+    When you need validation for some fields you can extend your data array with the required ones so that they will be validated.
```

Comment:
I actually do not understand this addition. This block talks about partial submission of data. Adding more data here seems to defeat the use of a PATCH request as long as I do not miss anything.

Reply:
The problem was that I had a `contraints` option in my `UpdateUserProfileType`.
This was not thrown an exception without that because of this clearMissing which not throws for not filled fields - as you can read here: https://github.com/symfony/symfony/issues/25331
So then I found out that moving this `constraints` option to the entity direct made it work's in the form.

So now instead of having...
```
->add('email', EmailType::class, ['constraints' => [
    new Assert\Email(['groups' => ['profile_update', 'user_registration']]),
]])
```
I have only this...
```
->add('email', EmailType::class)
```

And then my entity got's this.

```
/**
 * @var string
 *
 * @Assert\Email(groups={"profile_update", "user_registration"})
 */
protected $email;
```

And then it's working.

But I can not use the `constraints` options because it will not validate when the user not `PATCH`ing it. Thats the problem.
So my "moving contraints option to entity" solves the problem but not the root cause.

Reply:
Then I think we need to expand explanations on the `constraints` option instead to make it clear that constraints are ignored if the field they are bound to is not submitted.

Reply:
That's what I want to try with this commit with the **example** 😉
But now you understand what I want/mean.

Reply:
see #12365